### PR TITLE
Bug 1871732: Fix nodeInformer call in EgressIPManager.

### PR DESCRIPTION
**Pull request by danielmellado**

### CODE REVIEW

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now c1a193538ad0c12237a758b3551de5be735c96af

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now c1a193538ad0c12237a758b3551de5be735c96af

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 9528e1678c124b8b4b49fa7b3dac42db05949e62

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 7f8032b7cece300ec058262f86c1abd8e1fb7242

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 575599749bb9ae6eac9cf7e096c3f4ff4ff7a758

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 2505c6d061334b472e7065e05cc6dd3d604134bf

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now f7031bc46cbe72388a06cfcfa1317c528155c00c

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 52ae210cb9ca8b7f104a734badc9afee588c97fa

File: pkg/network/master/egressip.go [hunk 1]
```diff
@@ -163,6 +165,18 @@ func (eim *egressIPManager) poll(stop chan struct{}) {
 	}
 }
 
+func isNodeConditionGood(cond v1.NodeCondition) bool {
+	isNodeGood := true
+	if cond.Type == v1.NodeReady {
+		klog.Warningf("Node condition type is %s", cond.Type)
```
Comment (danwinship) [inline pkg/network/master/egressip.go:171]: debugging leftover? Don't log warnings about things that aren't warning-worthy...

File: pkg/network/master/egressip.go [hunk 2]
```diff
@@ -163,6 +165,18 @@ func (eim *egressIPManager) poll(stop chan struct{}) {
...
+		if cond.Status == v1.ConditionFalse || cond.Status == v1.ConditionUnknown {
+			klog.Warningf("Node condition status is %s", cond.Status)
+			isNodeGood = false
+		}
+	}
+	return isNodeGood
+}
```
Comment (danwinship) [inline pkg/network/master/egressip.go:178]: This is a confusing function. I'd change it to `nodeIsReady(node *v1.Node)`, and move the `range node.Status.Conditions` loop into here. Then you don't have the weirdness of claiming that any non-`NodeReady` condition counts as "good"

File: pkg/network/master/egressip.go [hunk 3]
```diff
@@ -177,18 +191,19 @@ func (eim *egressIPManager) check(retrying bool) (bool, error) {
 			continue
 		}
 
+		klog.Warningf("The node name is %s", node.name)
```
Comment (danwinship) [inline pkg/network/master/egressip.go:194]: another stray warning

File: pkg/network/master/egressip.go [hunk 4]
```diff
@@ -177,18 +191,19 @@ func (eim *egressIPManager) check(retrying bool) (bool, error) {
...
 		nn, err := eim.nodeInformer.Lister().Get(node.name)
 		if err != nil {
 			klog.Warningf("Couldn't get node %s ", node.ip)
```
Comment (danwinship) [inline pkg/network/master/egressip.go:197]: you need to return/break/continue/something here; as the code is now, it's going to try to dereference `nn` anyway.
also you should be including the error in the logged message

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now bc67633f872514395296fdc4ed7faa270583ce9f

File: pkg/network/master/egressip.go (hunk 2 above)
Comment (danielmellado) [reply pkg/network/master/egressip.go:181]: I've refactored it, will test on a cluster in any case.

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now fd82486750f97d583cd01bfe3e347195cb397416

File: pkg/network/master/egressip.go [hunk 5]
```diff
@@ -163,6 +164,18 @@ func (eim *egressIPManager) poll(stop chan struct{}) {
 	}
 }
 
+func nodeIsReady(node *v1.Node) bool {
+	nodeReady := true
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady {
+			if cond.Status == v1.ConditionFalse || cond.Status == v1.ConditionUnknown {
+				nodeReady = false
```
Comment (danwinship) [inline pkg/network/master/egressip.go:172]: you can just `return false` here (and `return true` at the end; you don't need the `nodeReady` variable)

**Comment by danielmellado** [comment]
https://github.com/openshift/sdn/pull/171

**Comment by danielmellado** [comment]
cherry-pick release-4.5

**Pull request merged**